File: app/src/main/java/net/chmielowski/fiszki/StartActivity.java
package net.chmielowski.fiszki;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.widget.SeekBar;
import android.widget.TextView;

public class StartActivity extends AppCompatActivity {

    private int numberOfWords = 3;
    private RealmDelegate realmDelegate = new RealmDelegate();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        realmDelegate.onCreate();
        setContentView(R.layout.activity_start);
        setLanguage(R.id.greek, DictionaryUtils.Lang.GREEK);
        setLanguage(R.id.italian, DictionaryUtils.Lang.ITALIAN);
        showMyWords(findViewById(R.id.words), this);
    }

    private void showMyWords(RecyclerView view, StartActivity context) {
        view.setLayoutManager(new LinearLayoutManager(context));
        view.setAdapter(new WordsAdapter(realmDelegate.getRealm()));
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        realmDelegate.onDestroy();
    }

    private void setLanguage(int button, DictionaryUtils.Lang lang) {
        findViewById(button).setOnClickListener(view -> {
            final Intent intent = new Intent(getApplicationContext(), MainActivity.class);
            intent.putExtra(MainActivity.LANGUAGE, lang);
            intent.putExtra(MainActivity.NUMBER_OF_WORDS, numberOfWords);
            startActivity(intent);
        });
        ((SeekBar) findViewById(R.id.numberOfWordsSeekBar))
                .setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
                    @Override
                    public void onProgressChanged(SeekBar seekBar, int progress, boolean b) {
                        numberOfWords = progress;
                        ((TextView) findViewById(R.id.numberOfWordsText))
                                .setText(String.valueOf(progress));
                    }

                    @Override
                    public void onStartTrackingTouch(SeekBar seekBar) {

                    }

                    @Override
                    public void onStopTrackingTouch(SeekBar seekBar) {

                    }
                });
    }

}

File: app/src/main/java/net/chmielowski/fiszki/StartActivity.kt
package net.chmielowski.fiszki

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.SeekBar
import android.widget.TextView
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers
import java.text.SimpleDateFormat
import java.util.*

class StartActivity : AppCompatActivity() {

    private var numberOfWords = 3
    private val realmDelegate = RealmDelegate()
    lateinit var roomService: RoomService

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        realmDelegate.onCreate()
        setContentView(R.layout.activity_start)
        setUp(R.id.start_lesson, DictionaryUtils.Lang.ITALIAN)
        roomService.data()
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe {
                    @SuppressLint("SimpleDateFormat")
                    val txt = it.map { lessonInfo -> lessonInfo.time }
                            .map { time -> SimpleDateFormat("yyyyy-mm-dd hh:mm:ss").format(Date(time)) }
                            .reduce { acc, s -> "$acc\n$s" }
                    findViewById<TextView>(R.id.last_lessons)
                            .setText(txt)
                }
    }

    override fun onDestroy() {
        super.onDestroy()
        realmDelegate.onDestroy()
    }

    private fun setUp(button: Int, lang: DictionaryUtils.Lang) {
        findViewById<View>(button).setOnClickListener {
            val intent = Intent(applicationContext, MainActivity::class.java)
            intent.putExtra(MainActivity.LANGUAGE, lang)
            intent.putExtra(MainActivity.NUMBER_OF_WORDS, numberOfWords)
            startActivity(intent)
        }
        (findViewById<View>(R.id.numberOfWordsSeekBar) as SeekBar)
                .setOnSeekBarChangeListener(object : SeekBar.OnSeekBarChangeListener {
                    override fun onProgressChanged(seekBar: SeekBar, progress: Int, b: Boolean) {
                        numberOfWords = progress
                        (findViewById<View>(R.id.numberOfWordsText) as TextView).text = progress.toString()
                    }

                    override fun onStartTrackingTouch(seekBar: SeekBar) {

                    }

                    override fun onStopTrackingTouch(seekBar: SeekBar) {

                    }
                })
    }

}
